### PR TITLE
ci: uploading artifacts on push

### PR DESCRIPTION
**Pull request by mnoman09**
## Summary
- Added steps to upload the build artifacts on s3

## Issues
- OASIS-8370

### CODE REVIEW

File: .github/workflows/csharp.yml [hunk 1]
```diff
@@ -6,41 +6,21 @@ on:
     branches: [master]
   pull_request:
     branches: [master]
+    
+env:
+  RELEASE_BRANCH: "3.3.x"
 
 jobs:
-  lint_code_base:
```
Comment (mikechu-optimizely) [inline .github/workflows/csharp.yml:11]: Are we removing linting?  I know it was failing but...

File: .github/workflows/csharp.yml [hunk 2]
```diff
@@ -6,41 +6,21 @@ on:
...
-          fetch-depth: 0
-      - name: Lint Code Base
-        uses: github/super-linter@v4
-        env:
-          VALIDATE_ALL_CODEBASE: false
-          DEFAULT_BRANCH: master
-          VALIDATE_CSHARP: true
-          VALIDATE_MARKDOWN: true
 
-  integration_tests:
```
Comment (mikechu-optimizely) [inline .github/workflows/csharp.yml:28]: I thought we'd want to run integration and FSC 🤔

**Review by mikechu-optimizely — COMMENTED** [review]
Just added some comments about the removals.

I'm glad to get the upload to S3 going.

**Comment by mikechu-optimizely** [comment]
We could add OASIS-8370 to the description to satisify the "Jira ticket reference check / jira_ticket_reference_check" if that's helpful

**Comment by mikechu-optimizely** [comment]
I'll leave the Approval to @msohailhussain

File: .github/workflows/csharp.yml (hunk 1 above)
Comment (mnoman09) [reply .github/workflows/csharp.yml:11]: Sorry that was a mistake. Fixed it

**Review by msohailhussain — APPROVED** [review]
lgtm

**Pull request merged**